catalog/core/catalog-core-standardframework/src/main/resources/OSGI-INF/blueprint/blueprint.xml

```diff
@@ -72,24 +72,44 @@
         <argument ref="catalogFramework"/>
     </bean>
 
-    <bean id="sourcePoller" class="ddf.catalog.util.impl.SourcePoller">
-        <cm:managed-properties persistent-id="ddf.catalog.util.impl.SourcePoller"
-                               update-strategy="container-managed"/>
-        <argument ref="sourcePollerRunner"/>
+    <bean id="sourceAvailabilityChecksThreadPool" class="java.util.concurrent.Executors"
+      factory-method="newCachedThreadPool">
+        <argument ref="sourcePollerScheduledPollingThreadFactory"/>
+    </bean>
+
+    <bean id="sourcePollerScheduledPollingThreadFactory"
+      class="org.codice.ddf.platform.util.StandardThreadFactoryBuilder"
+      factory-method="newThreadFactory">
+        <argument value="sourcePollerScheduledPollingThread"/>
+    </bean>
+
+    <bean id="pollAllSourcesThreadPool" class="java.util.concurrent.Executors"
```

Comment:
When we started using the google concurrency stuff, we started using this as a thread pool to kick off the polling thread, as well as the "watcher" threads that timeout our availability checks.

Reply:
Why not use the `availabilityChecksThreadPool` for the watcher threads and keep the ScheduledExecutorService as a single thread? I think one important advantage that the `newSingleThreadScheduledExecutor` has, compared to `newScheduledThreadPool`, is that it will make sure that the previously submitted task completes before starting a new task, which is what we want, right?

Reply:
That makes sense, but is it necessary since all our checks should be done by the timeout, which should always be shorter than the poll interval? I'd like to avoid injecting another executor service but I can if we need to.